File: 75. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) 
    {
        int zero = 0, one = 0, two = 0;
        for(auto val:nums)
        {
            if(val == 0) zero++;
            else if(val == 1) one++;
            else two++;
        }
        for(auto &val : nums)
        {
            if(zero) val = 0, zero--;
            else if(one) val = 1, one--;
            else val = 2;
        }
    }
};

File: Array/75. Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        zeroCount, oneCount, twoCount = 0, 0, 0
        for val in nums:
            zeroCount += val == 0
            oneCount += val == 1
            twoCount += val == 2
        
        cur = 0
        for _ in range(zeroCount):
            nums[cur] = 0
            cur += 1
        for _ in range(oneCount):
            nums[cur] = 1
            cur += 1
        for _ in range(twoCount):
            nums[cur] = 2
            cur += 1